charts/tidb-operator/templates/admission/jobPatch/post-del-secret-job.yaml

```diff
@@ -0,0 +1,34 @@
+{{- if and (eq .Values.admissionWebhook.enabled true) (eq .Values.admissionWebhook.certProvider "autoGenerator") }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "chart.name" . }}-admission-delete
+  labels:
+    app.kubernetes.io/name: {{ template "chart.name" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component:  admission-cert
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+"  "_" }}
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": post-delete
```

Comment:
With AA, we'll deprecate the Helm Chart management for TiDB cluster. But for tidb-operator, it's now more and more tightly coupled with Helm.
I'm wondering how much work we need to do in the future to make tidb-operator deployable with only `kubectl`.

Reply:
Here using `helm hook` is better but not necessary as each resources could retry until their dependency ready.